// react-native-ble-manager/dist/cjs/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectionPriority = exports.BleScanPhyMode = exports.BleScanMatchCount = exports.BleScanCallbackType = exports.BleScanMatchMode = exports.BleScanMode = exports.BleState = void 0;
/**
 * Android states: https://developer.android.com/reference/android/bluetooth/BluetoothAdapter#EXTRA_STATE
 * iOS states: https://developer.apple.com/documentation/corebluetooth/cbcentralmanagerstate
 * */
var BleState;
(function (BleState) {
    /**
     * [iOS only]
     */
    BleState["Unknown"] = "unknown";
    /**
     * [iOS only]
     */
    BleState["Resetting"] = "resetting";
    BleState["Unsupported"] = "unsupported";
    /**
     * [iOS only]
     */
    BleState["Unauthorized"] = "unauthorized";
    BleState["On"] = "on";
    BleState["Off"] = "off";
    /**
     * [Android only]
     */
    BleState["TurningOn"] = "turning_on";
    /**
     * [Android only]
     */
    BleState["TurningOff"] = "turning_off";
})(BleState || (exports.BleState = BleState = {}));
/**
 * [Android only]
 */
var BleScanMode;
(function (BleScanMode) {
    BleScanMode[BleScanMode["Opportunistic"] = -1] = "Opportunistic";
    BleScanMode[BleScanMode["LowPower"] = 0] = "LowPower";
    BleScanMode[BleScanMode["Balanced"] = 1] = "Balanced";
    BleScanMode[BleScanMode["LowLatency"] = 2] = "LowLatency";
})(BleScanMode || (exports.BleScanMode = BleScanMode = {}));
/**
 * [Android only]
 */
var BleScanMatchMode;
(function (BleScanMatchMode) {
    BleScanMatchMode[BleScanMatchMode["Aggressive"] = 1] = "Aggressive";
    BleScanMatchMode[BleScanMatchMode["Sticky"] = 2] = "Sticky";
})(BleScanMatchMode || (exports.BleScanMatchMode = BleScanMatchMode = {}));
/**
 * [Android only]
 */
var BleScanCallbackType;
(function (BleScanCallbackType) {
    BleScanCallbackType[BleScanCallbackType["AllMatches"] = 1] = "AllMatches";
    BleScanCallbackType[BleScanCallbackType["FirstMatch"] = 2] = "FirstMatch";
    BleScanCallbackType[BleScanCallbackType["MatchLost"] = 4] = "MatchLost";
})(BleScanCallbackType || (exports.BleScanCallbackType = BleScanCallbackType = {}));
/**
 * [Android only]
 */
var BleScanMatchCount;
(function (BleScanMatchCount) {
    BleScanMatchCount[BleScanMatchCount["OneAdvertisement"] = 1] = "OneAdvertisement";
    BleScanMatchCount[BleScanMatchCount["FewAdvertisements"] = 2] = "FewAdvertisements";
    BleScanMatchCount[BleScanMatchCount["MaxAdvertisements"] = 3] = "MaxAdvertisements";
})(BleScanMatchCount || (exports.BleScanMatchCount = BleScanMatchCount = {}));
/**
 * [Android only]
 */
var BleScanPhyMode;
(function (BleScanPhyMode) {
    BleScanPhyMode[BleScanPhyMode["LE_1M"] = 1] = "LE_1M";
    BleScanPhyMode[BleScanPhyMode["LE_2M"] = 2] = "LE_2M";
    BleScanPhyMode[BleScanPhyMode["LE_CODED"] = 3] = "LE_CODED";
    BleScanPhyMode[BleScanPhyMode["ALL_SUPPORTED"] = 255] = "ALL_SUPPORTED";
})(BleScanPhyMode || (exports.BleScanPhyMode = BleScanPhyMode = {}));
/**
 * [Android only API 21+]
 */
var ConnectionPriority;
(function (ConnectionPriority) {
    ConnectionPriority[ConnectionPriority["balanced"] = 0] = "balanced";
    ConnectionPriority[ConnectionPriority["high"] = 1] = "high";
    ConnectionPriority[ConnectionPriority["low"] = 2] = "low";
})(ConnectionPriority || (exports.ConnectionPriority = ConnectionPriority = {}));
